jme3-core/src/main/java/com/jme3/anim/tween/action/Action.java

```diff
@@ -1,22 +1,112 @@
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
 package com.jme3.anim.tween.action;
 
 import com.jme3.anim.AnimationMask;
+import com.jme3.anim.Armature;
 import com.jme3.anim.tween.Tween;
 import com.jme3.util.clone.Cloner;
 import com.jme3.util.clone.JmeCloneable;
 
+/**
+ * Base implementation (1st level impl/prototype) of the interface #{@link Tween} for the new animation system.
+ * The Action class encloses an array of {@link Tween} and converts them into jme animation actions if they actually aren't.
+ *
+ * NB : this implementation mimics the {@link com.jme3.anim.tween.AbstractTween}, but it delegates the interpolation method #{@link Tween#interpolate(double)}
+ * to the #{@link BlendableAction} class.
+ *
+ * Subclasses need to make a constructor call passing in the Tween Actions to enclose them & override #{@link Action#interpolate(double)}.
+ *
+ * Example of operation(Uses Abstract Factory Pattern Method(interpolate/doInterpolate)) :
+ *
+ * <li>#Check out #{@link BlendableAction} class.</li>
+ * <li>#Custom Code : </li>
+ * <pre class="prettyprint">
+ *  public abstract class AbstractAction extends Action {
+ *      private double length;
+ *      private volatile float interpolationDelta = 0f;
+ *      public ActionImpl(Tween... tweenActions){
+ *          super(tweenActions);
+ *      }
+ *      public boolean interpolate(double t){
+ *          //animation is out or in a reverse order.
+ *          if(t < 0){
+ *              return true;
+ *          }
+ *          //find where is the interpolation slider for transition.
+ *          //interpolationDelta would be -plugged into the interpolateTransforms(transformI, transformII, delta)- as a shared value.
+ *          interpolationDelta = t / length;
+ *          doInterpolate(t);
+ *          //interpolate as long as the interpolation slider haven't reached the full strength.
+ *          return interpolationDelta < 1.0d;
+ *      }
+ *      public double getLength(){
+ *           return length;
+ *      }
+ *      public void cloneFields(Cloner cloner, Object original) {
+ *         //deep clone fields
+ *      }
+ *      public Action jmeClone() {
+ *          //shallow clone
+ *         try {
+ *             return (Action) super.clone();
+ *         } catch (CloneNotSupportedException exception) {
+ *             throw new RuntimeException(exception);
+ *         }
+ *     }
+ *      //override this factory method to do interpolation
+ *      protected abstract void doInterpolate(double t);
+ *  }
+ * </pre>
+ * <b>Created by Nehon.</b>
+ */
 public abstract class Action implements JmeCloneable, Tween {
 
     protected Action[] actions;
     private double length;
+    //the default speed is 1, which results in a normal tpf.
```

Comment:
At the `AnimComposer.Layer` class, this formula is applied on the time per frames, before interpolating : 

`time += tpf * currentAction.getSpeed() * ac.globalSpeed`

if the speed is 1 then (time = time + tpf), so here tpf shows a normal value and not a scaled value.

I know its somehow frustrating to address everything, if you have a better alias let me know.

Reply:
A quick look on the Tween interface docs shows : 
```java
 /**
     * Sets the implementation specific interpolation to the
     * specified 'tween' value as a value in the range from 0 to
     * getLength().  If the value is greater or equal to getLength()
     * then it is internally clamped and the method returns false.
     * If 't' is still in the tween's range then this method returns
     * true.
     *
     * @param t animation time (in de-scaled seconds)
     * @return true if t&gt;length(), otherwise false
     */
    boolean interpolate(double t);
```
t is a de-scaled value unless its scaled using the `action.speed` attribute.

Reply:
So `speed` doesn't affect `tpf` in any way. Instead of "a normal tpf" say something like "animation time = real time" or "animation clips run at their designed speed".

Reply:
I will change `normal tpf` to `animation clips run at their normal speed`.